game/handler.go

```diff
@@ -435,8 +442,86 @@ func handleConfigure(w ResponseWriter, r Request) error {
 	return nil
 }
 
+func handleFixNewTimestamps(w ResponseWriter, r Request) error {
+	dryRun := r.Req().URL.Query().Get("dry-run") != "false"
+
+	ctx := appengine.NewContext(r.Req())
+
+	user, ok := r.Values()["user"].(*auth.User)
+	if !ok {
+		return HTTPErr{"unauthenticated", 401}
+	}
+
+	superusers, err := auth.GetSuperusers(ctx)
+	if err != nil {
+		return err
+	}
+
+	if !superusers.Includes(user.Id) {
+		return HTTPErr{"unauthorized", 403}
+	}
+
+	gameIDs, err := datastore.NewQuery(gameKind).KeysOnly().GetAll(ctx, nil)
+	if err != nil {
+		return err
+	}
+
+	for _, gameID := range gameIDs {
+		if err := datastore.RunInTransaction(ctx, func(ctx context.Context) error {
+			game := &Game{}
+			if err := datastore.Get(ctx, gameID, game); err != nil {
+				return err
+			}
+			if game.StartedAt.IsZero() {
+				phases := Phases{}
+				phaseIDs, err := datastore.NewQuery(phaseKind).Ancestor(gameID).GetAll(ctx, &phases)
+				if err != nil {
+					return nil
+				}
+				if len(phases) > 0 {
+					keys := []*datastore.Key{gameID}
+					values := []interface{}{game}
+					sort.Sort(phases)
+					for i := range phases {
+						phase := &phases[i]
+						if phase.PhaseOrdinal != int64(i) {
+							return fmt.Errorf("WTF, the phases aren't sorted properly? Phase %v is %+v", i, phase)
+						}
+						if phase.CreatedAt.IsZero() {
+							// If this is the first phase OR DeadlineAt - lastPhase.CreatedAt > phase length.
+							if i == 0 || phase.DeadlineAt.Sub(phases[i-1].CreatedAt) > time.Minute*game.PhaseLengthMinutes {
+								phase.CreatedAt = phase.DeadlineAt.Add(-time.Minute * game.PhaseLengthMinutes)
+								log.Infof(ctx, "Updating phase of game with phase length %v, with DeadlineAt %v and no previous phase within %v to have CreatedAt %v", time.Minute*game.PhaseLengthMinutes, phase.DeadlineAt, time.Minute*game.PhaseLengthMinutes, phase.CreatedAt)
+							} else {
+								phase.CreatedAt = phase.DeadlineAt
```

Comment:
I don't understand this. Is this the case when there are no actions needed by players (e.g. Spring 1901 retreat).

Reply:
Exactly.

If all players are RDY - which happens if they have no Options, NMRed last phase, or are eliminated - the phase resolved immediately and gets a DeadlineAt when created.